lua/metals/setup.lua

```diff
@@ -0,0 +1,13 @@
+--[[
+  Facilities to help modularize the metals plugin config
+--]]
+
+local M = {}
+
+--[[ The only way to enable didSave with current LSP and metals
+capabilities is to override after init. --]]
+M.on_init = function(client, _)
```

Comment:
hmmm, not needed for this pr, but is this `on_init` something we could also add in all the `initializationOptions` that we are overriding?

Reply:
I didn't want to change too much here, but moving forward it'll depend on the config strategy. IMO we should hide this from the user as we talked before and accept some overrides from them, so the setup function internals are hidden from the user. IMO, nvim-lsp doesn't even add much value for nvim-metals, and we could have a simply add nvim-metals and works with defaults or has a single call to setup_metals, accepting overrides from the user. Then the nvim-lsp setup function is left for DYI config. That would allow more freedom to change things in nvim-metals, though this is a topic for further discussions.

Reply:
> hide this away from the user

If we went this route and actually put all the `initializationOptions` in here as well, that would _hide it away_ though right?

And I get what you're saying about the value of `nvim-lsp`. At this moment it basically boils down to me needing to fully understand how to start and manage the client. Once I get there, I'll go that route and just manage installation and everything probably in this plugin.